servo/connectors/kubernetes.py

```diff
@@ -4100,6 +4116,13 @@ class BaseKubernetesConfiguration(servo.BaseConfiguration):
         description="Time interval to wait before considering Kubernetes operations to have failed."
     )
 
+    @pydantic.validator("on_failure")
+    def validate_failure_mode(cls, v):
+        if v == FailureMode.destroy:
+            servo.logger.warning(f"Deprecated value 'destroy' used for 'on_failure', replacing with 'shutdown'")
+            return FailureMode.shutdown
+        return v
```

Comment:
Please add a unit test for this. It should probably go here (under TestKubernetesConfiguration):  https://github.com/opsani/servox/blob/0deeb2d5b7af7f0f529e52b28670db84b974f173/tests/connectors/kubernetes_test.py#L396

Something like the following should work:

```
    def test_failure_mode_destroy(self) -> None:
        config = servo.connectors.kubernetes.KubernetesConfiguration(
            namespace="default",
            description="Update the namespace, deployment, etc. to match your Kubernetes cluster",
            on_failure=servo.connectors.kubernetes.FailureMode.destroy,
            deployments=[
                servo.connectors.kubernetes.DeploymentConfiguration(
                    name="fiber-http",
                    replicas=servo.Replicas(
                        min=1,
                        max=2,
                    ),
                    containers=[
                        servo.connectors.kubernetes.ContainerConfiguration(
                            name="fiber-http",
                            cpu=servo.connectors.kubernetes.CPU(
                                min="250m", max="4000m", step="125m"
                            ),
                            memory=servo.connectors.kubernetes.Memory(
                                min="128MiB", max="4.0GiB", step="128MiB"
                            ),
                        )
                    ],
                )
            ],
        )
        assert config.on_failure == FailureMode.shutdown
        assert config.deployments[0].on_failure == FailureMode.shutdown
```